core/templates/pages/preferences-page/preferences-page.component.spec.ts

```diff
@@ -97,11 +100,13 @@ describe('Preferences Controller', function() {
     var isRequestTheExpectOne = function(queryParams) {
       return decodeURIComponent(queryParams).match('"update_type":"user_bio"');
     };
+    spyOn(PreventPageUnloadEventService, 'removeListener').and.callThrough();
 
     $httpBackend.expect(
       'PUT', '/preferenceshandler/data', isRequestTheExpectOne).respond(200);
     ctrl.saveUserBio(userBio);
     $httpBackend.flush();
+    expect(PreventPageUnloadEventService.removeListener).toHaveBeenCalled();
```

Comment:
Please add a separate test for this instead of adding this to the existing test block

Reply:
DONE